libsolidity/ast/Types.cpp

```diff
@@ -964,10 +964,38 @@ TypeResult RationalNumberType::binaryOperatorResult(Token _operator, Type const*
 {
 	if (_other->category() == Category::Integer || _other->category() == Category::FixedPoint)
 	{
-		auto commonType = Type::commonType(this, _other);
-		if (!commonType)
-			return nullptr;
-		return commonType->binaryOperatorResult(_operator, _other);
+		if (isFractional())
+			return TypeResult::err("Fractional literals not supported.");
+		else if (!integerType())
+			return TypeResult::err("Literal too large.");
```

Comment:
Why does not integer imply too large?

Reply:
@leonardoalt "@returns the smallest integer type that can hold the value or an empty pointer if not possible"